compiler/crates/common/src/feature_flags.rs

```diff
@@ -17,6 +17,7 @@ use serde::Serialize;
 
 use crate::Rollout;
 
+/// **NOTE**: Every change on this struct should be reflected into the `relay-compiler/index.d.ts` file.
```

Comment:
I wonder if we can make use of something like this: https://github.com/Aleph-Alpha/ts-rs here?

We would generate these typescript definitions based on the actual rust types?

Reply:
I did some POCs with [specta](https://docs.rs/specta/latest/specta/) yesterday, it seems to works well. I think that we can bring it to these structs too.

They have just a limitation around some JSDocs comments, like the `@default` and others, but it's working.

Reply:
Nice, @noghartt! I think this would be a much better solution to always keep these definitions in sync.

Reply:
@noghartt Just checking in here. Were you able to make any progress with Spectra? Do you think it can offer a viable automated solution?

Reply:
> @noghartt Just checking in here. Were you able to make any progress with Spectra? Do you think it can offer a viable automated solution?

I think that we can bring an automated solution yes. But as I commented, specta doesn't seem to fit all the types for us, as an example, it doesn't fill the JSDocs related fields, like the `@default`. I couldn't find a workaround for that scenario.

Except that, it seems to works well. Do you think that we can bring this improvement in this PR or maybe moving to another?